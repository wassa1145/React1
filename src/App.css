body {
  background-color: #f8f8f8;
  font-family: Arial, Helvetica, sans-serif;
}
.page {
  max-width: 1440px;
  min-width: 320px;
  margin: 0 auto;
  box-sizing: border-box;
}
.content {
  min-width: 445px;
  max-width: 600px;
  margin: 0 auto;
  background-color: #e4ebf8;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  height: 100vh;
}
