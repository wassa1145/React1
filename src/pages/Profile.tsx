import React, { FC, useContext, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { ThemeContext } from './../utils/ThemeContext';
import { changeName, toggleProfile } from '../store/profile/actions';
import { Button, TextField } from '@mui/material';
import { selectName, selectVisible } from './../store/profile/selectors';

export const Profile: FC = () => {
  const { theme, toggleTheme } = useContext(ThemeContext);
  const dispatch = useDispatch();
  const [value, setValue] = useState('');

  const visible = useSelector(selectVisible);
  const name = useSelector(selectName);

  return (
    <>
      <h2>Profile</h2>
      <div className="box">
        <div className="profile-row">
          <span>–¢–µ–º–∞: {theme === 'light' ? 'üåû –°–≤–µ—Ç–ª–∞—è' : 'üåô –¢–µ–º–Ω–∞—è'}</span>
          <Button variant="outlined" onClick={toggleTheme}>
            –ò–∑–º–µ–Ω–∏—Ç—å —Ç–µ–º—É
          </Button>
        </div>
      </div>
      <div className="box">
        <p>–ò–º—è: {name}</p>
        <div className="profile-row">
          <TextField
            variant="outlined"
            onChange={(e) => setValue(e.target.value)}
            placeholder="–í–≤–µ–¥–∏—Ç–µ –∏–º—è"
            value={value}
          />
          <Button
            variant="outlined"
            onClick={() => dispatch(changeName(value))}
          >
            –ò–∑–º–µ–Ω–∏—Ç—å –∏–º—è
          </Button>
        </div>
      </div>
      <div className="box">
        <input type="checkbox" checked={visible} />
        <Button variant="outlined" onClick={() => dispatch(toggleProfile())}>
          change visible
        </Button>
      </div>
    </>
  );
};
