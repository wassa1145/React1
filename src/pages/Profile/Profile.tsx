import React, { FC, useContext, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { ThemeContext } from 'src/utils/ThemeContext';
import { Button, TextField } from '@mui/material';
import { selectName, selectVisible } from 'src/store/profile/selectors';
import { changeName, toggleProfile } from 'src/store/profile/slice';
import { onValue, update } from 'firebase/database';
import { userRef } from 'src/services/firebase';

export const Profile: FC = () => {
  const { theme, toggleTheme } = useContext(ThemeContext);
  const dispatch = useDispatch();
  const [value, setValue] = useState('');

  const visible = useSelector(selectVisible);
  const name = useSelector(selectName);

  useEffect(() => {
    onValue(userRef, (snapshot) => {
      const user = snapshot.val();
      dispatch(changeName(user.name || ''));
    });
  }, []);
  const handleChangeName = async () => {
    update(userRef, {
      name: value,
    });
  };
  return (
    <>
      <h2>Profile</h2>
      <div className="box">
        <div className="profile-row">
          <span>–¢–µ–º–∞: {theme === 'light' ? 'üåû –°–≤–µ—Ç–ª–∞—è' : 'üåô –¢–µ–º–Ω–∞—è'}</span>
          <Button variant="outlined" onClick={toggleTheme}>
            –ò–∑–º–µ–Ω–∏—Ç—å —Ç–µ–º—É
          </Button>
        </div>
      </div>
      <div className="box">
        <p>–ò–º—è: {name}</p>
        <div className="profile-row">
          <TextField
            variant="outlined"
            onChange={(e) => setValue(e.target.value)}
            placeholder="–í–≤–µ–¥–∏—Ç–µ –∏–º—è"
            value={value}
          />
          <Button variant="outlined" onClick={handleChangeName}>
            –ò–∑–º–µ–Ω–∏—Ç—å –∏–º—è
          </Button>
        </div>
      </div>
      <div className="box">
        <input type="checkbox" checked={visible} readOnly={true} />
        <Button variant="outlined" onClick={() => dispatch(toggleProfile())}>
          change visible
        </Button>
      </div>
    </>
  );
};
