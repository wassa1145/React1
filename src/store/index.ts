import { compose, combineReducers } from 'redux';
import { profileReducer } from './profile/slice';
import { chatReducer } from './chats/reducer';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';

export const composeEnhancers =
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export type StoreState = ReturnType<typeof rootReducer>;

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['profile'],
};

const rootReducer = combineReducers({
  chats: chatReducer,
  profile: profileReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== 'production',
  middleware: [thunk],
});

export const persistor = persistStore(store);
