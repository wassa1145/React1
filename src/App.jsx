import React from 'react';
import './App.css';
import { Form } from './components/FormFunc/Form';

export const App = () => {
  return <>
    <Form />
  </>
}

