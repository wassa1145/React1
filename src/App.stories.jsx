import React from 'react';
// import { within, userEvent } from '@storybook/testing-library';

import { App } from './App';

export default {
  title: 'MyComponents/Page',
  component: App,
};

const Template = (args) => <App {...args} />;

export const EmptyPage = Template.bind({});

// с play падают тесты

// export const InteractionPage = Template.bind({});
// InteractionPage.play = async ({ canvasElement }) => {
//   const canvas = within(canvasElement);
//   await userEvent.type(canvas.getByTestId('textarea'), 'Какой-то текст', {
//     delay: 100,
//   });
//   const submitButton = await canvas.getByTestId('submit-button');
//   await userEvent.click(submitButton);
// };
