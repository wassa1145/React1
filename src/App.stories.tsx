import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { App } from './App';
import { MemoryRouter } from 'react-router-dom';

export default {
  title: 'MyComponents/Page',
  component: App,
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = (args) => (
  <MemoryRouter>
    <App {...args} />
  </MemoryRouter>
);

export const EmptyPage = Template.bind({});
