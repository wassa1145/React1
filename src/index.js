import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import { App } from './App'; // указываем App в фигурных скобках, т.к к нас не дефоллтный импорт а именованный,
// если бы был дефолтный экспорт в компоненте App.js то импорт был бы тоже дефолтный без {}


ReactDOM.render(
  <App />,
  document.getElementById('root')
);

