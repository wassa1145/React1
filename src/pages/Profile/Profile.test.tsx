import React, { useState } from 'react';
import { Profile } from './Profile';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from 'src/store';
import userEvent from '@testing-library/user-event';
import { defaultContext, ThemeContext } from 'src/utils/ThemeContext';

describe('App', () => {
  it('render with snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('change name', async () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    await userEvent.type(screen.getByRole('textbox'), 'new name');
    await userEvent.click(
      screen.getAllByRole('button', { name: /Изменить имя/ })[0]
    );
    expect(screen.getByText(/new name/)).toBeInTheDocument();
  });

  it('toggle theme', async () => {
    const ThemeProfile = () => {
      const [theme, setTheme] = useState(defaultContext.theme);
      const toggleTheme = () => {
        setTheme(theme === 'light' ? 'dark' : 'light');
      };
      return (
        <ThemeContext.Provider
          value={{
            theme,
            toggleTheme,
          }}
        >
          <Provider store={store}>
            <Profile />
          </Provider>
        </ThemeContext.Provider>
      );
    };
    render(<ThemeProfile />);

    expect(screen.getByText(/Светлая/)).toBeInTheDocument();
    await userEvent.click(
      screen.getAllByRole('button', { name: /Изменить тему/ })[0]
    );
    expect(screen.getByText(/Темная/)).toBeInTheDocument();
  });
});
