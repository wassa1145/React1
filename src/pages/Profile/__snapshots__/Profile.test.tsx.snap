// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render with snapshot 1`] = `
<DocumentFragment>
  <h2>
    Profile
  </h2>
  <div
    class="box"
  >
    <div
      class="profile-row"
    >
      <span>
        –¢–µ–º–∞: üåû –°–≤–µ—Ç–ª–∞—è
      </span>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        –ò–∑–º–µ–Ω–∏—Ç—å —Ç–µ–º—É
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="box"
  >
    <p>
      –ò–º—è: default name
    </p>
    <div
      class="profile-row"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="–í–≤–µ–¥–∏—Ç–µ –∏–º—è"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ‚Äã
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        –ò–∑–º–µ–Ω–∏—Ç—å –∏–º—è
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="box"
  >
    <input
      checked=""
      readonly=""
      type="checkbox"
    />
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      change visible
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
